### ISSUE TITLE
Windows are moved, when screen locks

### ISSUE THREAD

**Issue by oleid** [zakkak/workspace-grid#5]
Thanks for maintaining the extension. It works for me better than before. However, I found a new issue:

I've a 3x3 virtual desktop layout. It works great during normal work, but as soon as I unlock the screen when it got locked due to inactivity, the windows are moved as following:
- the windows in the 3rd row are moved to the central desktop
- the windows in the 2nd row 3rd column are moved to the central desktop, too.
- the windows in the 1st row 3rd column aren't moved at all, they stay where they are.

So basically desktop 6 to 9 are empty after a screen lock, all the windows are moved to desktop 5.

I'm on GNOME-3.14, as provided by ArchLinux.

**Comment by danbernier** [comment]
Any progress on this? I'm still seeing this behavior on Ubuntu 16.04.

**Comment by zakkak** [comment]
No, but I think you can workaround the issue by setting the number of workspaces in gnome to the same as in the workspace grid.

**Comment by squarewav** [comment]
I'm seeing an issue that is probably something different entirely but if I were to create a new issue it would have the same exact title so I'll post here for now.

I have a laptop but use an external display. If the external monitor sleeps, after unlocking the screen, all windows have been moved to the first column.

I have been using Linux since kernel 2.0.35 and it is very sad to say that I have been running into variations of this issue since the beginning. X insists that applications save their position. So by pushing this responsibility into applications, different applications have various levels of success at recalling their state and geometry. It's really absurd to be tripping over the same issue for 20+ years!

**Comment by kyle-jennings** [comment]
Seeing the issue on Arch (and previously Ubuntu 17.10) with Wayland+Gnome3.  I have a 2x3 sized grid:
| 1 | 2 | 3 |
| 4 | 5 | 6 |

Every time I lock the screen, all my windows on the bottom row will move over to workspace 4.  All the top rows seem fine.

**Comment by zakkak** [comment]
Did you try setting the number of workspaces in gnome to the same as in the workspace grid?

**Comment by kyle-jennings** [comment]
Yes, in gnome tweaks I have the "number of workspaces" set to 6, and then in the Workspace Grid extension I have rows set to 2 and columns set to 3

**Comment by zakkak** [comment]
And I assume you are using static workspaces. I don't know what could be wrong. In most cases it had to do with the configuration.

**Comment by kyle-jennings** [comment]
Yeah im using static workspaces as well.  Are there some logs I should be looking at? I havent had a chance to really give this a look over just yet.

**Comment by kyle-jennings** [comment]
Looks like an update was just rolled out.  I installed it and restarted my machine and BOOM.  it works now!

**Comment by zakkak** [comment]
The update just adds support (updates the metadata file) for the latest gnome version. I guess something was wrong with your installation and updating somehow fixed it. It is strange however, that you had the issue on more than one OSes (unless you did not format your home directory).

**Comment by tjaalton** [comment]
I hit this same issue with a fresh install of ubuntu bionic; I had a 2x2 grid first, but changed it later to 2x3. Fixed it by disabling/enabling the extension, the number of workspaces under Workspaces was already correct so didn't have to touch that anymore.

**Comment by pbass04** [comment]
I had this problem until I looked through this thread and found that zakkak had inferred that the extension was turning off when the computer was locked, thus moving all of my windows into the "last" workspace which was number 4 for me. The fix for me was as follows.

1) Open Tweaks
2) Click Extensions
3) Turn off the Workspace grid extension
4) Click Workspaces
5) Make sure the Static Workspaces was checked.
6) Made the Number of Workspaces 9
7) Click Extensions again
8) Turn Workspace grid back on.

I tested locking the computer and unlocking it and all of my windows are in their proper grid spaces using a 3 column 3 row configuration.

**Comment by BjarniRunar** [comment]
Hello! This is still a problem on whatever ships with Ubuntu 18.04.

Why was this issue closed? That's just telling everyone this will never get fixed, is that the correct message? The bug has not been identified and it's extremely frustrating. By which I mean "throw my laptop out the window and then install KDE on a toaster" level frustrating.

My thanks to all the people posting workarounds. I will try them, but that doesn't change the fact that this is a bug that needs fixing. If it's actually a bug in some other extension, that's a totally legit conclusion, but then we want to go chat with the devs of that extension and get it fixed over there, right?

**Comment by BjarniRunar** [comment]
An update: It looks like the posted workarounds do work for me, unless I have the shell-tile extension enabled, in which case windows still jump around on screen lock.

So for any late-comers experiencing this, beware that shell-tile appears to make things worse.

As a suggestion for Workspace Grid, I think the "correct" solution here would be to refuse to allow configurations that are in conflict with the main GNOME settings. That would require erroring out and refusing to persist an NxM grid setting, where NxM != configured static workspace. It makes no sense to allow the user to configure something that cannot work, IMO.

That aside, thanks for the extension! Now that I've figured out how to configure it, it makes my desktop much more pleasant to use.
